src/modules/integrations/hostctl.nix

```diff
@@ -0,0 +1,46 @@
+{ pkgs, lib, config, ... }:
+
+let
+  hostContent = lib.concatStringsSep "\n" config.hostctl.hosts;
+  hostHash = builtins.hashString "sha256" hostContent;
+  file = pkgs.writeText "hosts" ''
+    # ${hostHash}
+    ${hostContent}
+  '';
+in
+{
+  options.hostctl = {
+    enable = lib.mkEnableOption "Adjust /etc/hosts using hostctl.";
+
+    profile = lib.mkOption {
+      type = lib.types.str;
+      default = "devenv";
```

Comment:
What happens if you have two devenvs setting different hosts?

Reply:
The last one wins. That's why I added the profileName option 😅

Reply:
What if we use `"devenv-${hostHash"` for the profile name?

Reply:
It needs to be something unique to this devenv to update also the area in the hosts

Reply:
`"devenv-${builtins.hashString "sha256" env.DEVENV_ROOT}"`